suite/tests/client-interface/set-stolen-reg-mcontext.c

```diff
@@ -0,0 +1,40 @@
+/* **********************************************************
+ * Copyright (c) 2020 Google, Inc.  All rights reserved.
+ * Copyright (c) 2020 Arm Limited   All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Google, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+#include <stdio.h>
+
+int main(void){
+    // just an instruction to detect in *.dll.c
+    __asm__ __volatile__("mov x27, #1" : : : "x27");
```

Comment:
I wasn't sure about checking for he stolen reg value in the app, as this fix is specifically about how the mcontext is maintained. For me I don't think the issue wast the app having a corrupt x28 register in its execution, but rather the client not knowing what the true value was.

Reply:
The suggestion to possibly have the app check the value is for dr_set_mcontext where the client is changing the app value.  It is admittedly not a very normal thing for a real tool to do, but a cooperative app in a test is ok.  Another suggestion is to do what the existing cleancall tests do: consecutive clean calls where one does a dr_set_mcontext and the next does a dr_get_mcontext and looks for the new value.

Reply:
I have implemented the latter here. Can do a cooperative app-client test instead if that's best.